File: src/java/2_CreationalPatterns/2_Factories/FactoryExercise.java
package com.activemesa.creational.factories.exercise;

class Person
{
  public int id;
  public String name;

  public Person(int id, String name)
  {
    this.id = id;
    this.name = name;
  }
}

class PersonFactory
{
  private int id = 0;

  public Person createPerson(String name)
  {
    return new Person(id++, name);
  }
}

File: src/js/2_CreationalPatterns/2_Factories/factory_exercise.js
class Person
{
  constructor(id, name)
  {
    this.id = id;
    this.name = name;
  }
}

class PersonFactory
{
  createPerson(name)
  {
    return new Person(
      PersonFactory.id++,
      name
    );
  }
}
PersonFactory.id = 0;

describe('factory', function()
{
  it('exercise', function()
  {
    let pf = new PersonFactory();

    let p1 = pf.createPerson('Chris');
    expect(p1.name).toEqual('Chris');
    expect(p1.id).toEqual(0);

    let p2 = pf.createPerson('Sarah');
    expect(p2.id).toEqual(1);
  });
});